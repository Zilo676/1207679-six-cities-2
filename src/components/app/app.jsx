import React from 'react';
import {Switch, Route} from 'react-router-dom';

import MainPage from '../main-page/main-page.jsx';
import {SignIn} from '../sign-in/sign-in.jsx';
import OfferDetails from '../offer-details/offer-details.jsx'

import withSignIn from '../../hocs/with-sign-in/with-sign-in';

const App = () => {

  const SignInWrapped = withSignIn(SignIn);

  return (
    <Switch>
      <Route exact path='/login' component={SignInWrapped} />
      <Route exact path='/offer/:id' render={({match}) => <OfferDetails id={+match.params.id} />} />
      <Route exact path='/' component={MainPage} />
    </Switch>
  );
};

export {App};
