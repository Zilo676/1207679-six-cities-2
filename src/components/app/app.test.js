import React from 'react';
import renderer from 'react-test-renderer';
import {App} from './app.jsx';

const createNodeMock = () => document.createElement(`div`);

jest.mock(`../main-page/main-page.jsx`, () => `Main Page`);

it(`Correctly rendered`, () => {
  const tree = renderer.create(<App />, {createNodeMock}).toJSON();
  expect(tree).toMatchSnapshot();
});
