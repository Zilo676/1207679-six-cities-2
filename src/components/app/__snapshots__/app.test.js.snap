// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correctly rendered 1`] = `<Main Page />`;
